spec/features/shopping_list_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+describe '/general_shopping_list/:recipe_id', type: :feature do
+  let(:user) { FactoryBot.create(:user) }
+
+  before do
+    user.confirm # Confirm the user's email address
+    visit new_user_session_path
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Log in'
+    expect(page).to have_text('Signed in successfully.')
+  end
+
+  # let!(:food) { FactoryBot.create(:food, user:) }
+  # let!(:food2) { FactoryBot.create(:food, user:) }
```

Comment:
- [ OPTIONAL] 
Kindly check your codebase and remove unused or commented pieces of code. This makes your codebase clean and professional.